Manager/src/main/java/io/raspberrywallet/manager/bitcoin/Bitcoin.java

```diff
@@ -328,6 +338,34 @@ public void sendCoins(String amount, String recipient) throws WalletNotInitializ
         }
     }
 
+    /**
+     * @return list of transactions related with this wallet
+     */
+    public List<TransactionView> getAllTransactions() throws WalletNotInitialized {
+        final Wallet wallet = getWallet();
+        final ArrayList<Transaction> transactions = new ArrayList<>(wallet.getTransactions(false));
+
+        return transactions.stream().map(tx -> {
+                    TransactionView txView = new TransactionView();
+                    txView.setTxHash(tx.getHashAsString());
+                    txView.setCreationTimestamp(tx.getUpdateTime().getTime());
+
+                    final Address toAddress = tx.getOutput(0).getAddressFromP2PKHScript(TestNet3Params.get());
+                    if (toAddress != null) txView.setToAddress(toAddress.toString());
+
+                    txView.setAmountFromMe(tx.getValueSentFromMe(wallet).toFriendlyString());
+                    txView.setAmountToMe(tx.getValueSentToMe(wallet).toFriendlyString());
+
+                    long fee = (tx.getInputSum().getValue() > 0 ? tx.getInputSum().getValue() - tx.getOutputSum().getValue() : 0);
```

Comment:
maybe it should be as a new method? i'm not sure what going on here and method name would explain it, also let's avoid conditional operator ( ? : ), because expression is too complicated 
```java
private long calculateSomething(Object tx) {
    if (tx.getInputSum().getValue() > 0)
        return tx.getInputSum().getValue() - tx.getOutputSum().getValue();
    else
        return 0;
}
```

Reply:
I replaced it with Math.max(0, fee)